lib/segment/src/index/field_index/geo_index/mutable_geo_index.rs

```diff
@@ -49,6 +54,136 @@ impl MutableGeoMapIndex {
             db,
             store_cf_name,
         ));
+        Self {
+            dynamic_index: DynamicGeoMapIndex::new(),
+            db_wrapper,
+        }
+    }
+
+    pub fn db_wrapper(&self) -> &DatabaseColumnScheduledDeleteWrapper {
+        &self.db_wrapper
+    }
+
+    pub fn load(&mut self) -> OperationResult<bool> {
+        if !self.db_wrapper.has_column_family()? {
+            return Ok(false);
+        };
+
+        let mut points_to_hashes: BTreeMap<PointOffsetType, Vec<GeoHash>> = Default::default();
+
+        for (key, value) in self.db_wrapper.lock_db().iter()? {
+            let key_str = std::str::from_utf8(&key).map_err(|_| {
+                OperationError::service_error("Index load error: UTF8 error while DB parsing")
+            })?;
+
+            let (geo_hash, idx) = GeoMapIndex::decode_db_key(key_str)?;
+            let geo_point = GeoMapIndex::decode_db_value(value)?;
+
+            if self.dynamic_index.point_to_values.len() <= idx as usize {
+                self.dynamic_index
+                    .point_to_values
+                    .resize_with(idx as usize + 1, Vec::new);
+            }
+
+            if self.dynamic_index.point_to_values[idx as usize].is_empty() {
+                self.dynamic_index.points_count += 1;
+            }
+
+            points_to_hashes.entry(idx).or_default().push(geo_hash);
+
+            self.dynamic_index.point_to_values[idx as usize].push(geo_point);
+            self.dynamic_index
+                .points_map
+                .entry(geo_hash)
+                .or_default()
+                .insert(idx);
+
+            self.dynamic_index.points_values_count += 1;
+        }
+
+        for (_idx, geo_hashes) in points_to_hashes {
+            self.dynamic_index.max_values_per_point =
+                max(self.dynamic_index.max_values_per_point, geo_hashes.len());
+            self.dynamic_index.increment_hash_point_counts(&geo_hashes);
+            for geo_hash in geo_hashes {
+                self.dynamic_index.increment_hash_value_counts(&geo_hash);
+            }
+        }
+        Ok(true)
+    }
+
+    pub fn remove_point(&mut self, idx: PointOffsetType) -> OperationResult<()> {
+        if let Some(removed_geo_points) = self.dynamic_index.point_to_values.get(idx as usize) {
+            for removed_geo_point in removed_geo_points {
+                let removed_geo_hash: GeoHash =
```

Comment:
Naming nit: I think the names `geo_points_to_remove` and `geo_hash_to_remove` would be less confusing.

Reply:
Fixed for all functions in geo index

Reply:
Fixed